wordpress-mobile/WordPress-Android
# Plans on JP:  Remove feature flag

Fixes #19155 

Remove plans feature flag, as it is a stable feature now.

<img width=320 src="https://github.com/wordpress-mobile/WordPress-Android/assets/990349/a48b00b3-3714-4c2c-9918-104142bf5ea1" />

To test:

- Launch JP mobile app
- `Choose` a site that is not a P2, with a Free plan,  and in admin role
- `Verify` it shows Plans card as before
- `Open` debug settings (Me -> Debut settings)
- `Verify` feature flag `dashboard_card_free_to_paid_plans` is not present

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Existing Unit and UI tests

PR submission checklist:
